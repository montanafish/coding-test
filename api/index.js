export const getShowTitles = (numOfRecords = 1) => {
  const showTitles = []

  for (let i = 0; i < numOfRecords; i += 1) {
    showTitles.push({
      id: i,
      title: 'Home and Away',
      description: 'THE ENDLESS SUMMER',
      image: {
        medium: '/static/images/tile-medium.jpg',
        large: '/static/images/tile.jpg'
      },
      icon: '/static/images/logo-40x40.png',
      seasons: 8,
      views: 5752532,
    })
  }
  return showTitles
}
