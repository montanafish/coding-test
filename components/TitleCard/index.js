import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { colors } from 'styles/variables';

const Box = styled.div`
  position: relative;
`;

const BoxLink = styled.a`
  &:hover {
    opacity: 0.8;
  }
`;

const BoxBar = styled.div`
  position: absolute;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  color: ${colors.white};
  width: 100%;
  padding: 10px;
  height: 0.33;
`;

const BoxArt = styled.img`
  width: 100%;
`;

const Icon = styled.img`
  max-width: 40px;
  padding-right: 10px;
`;

const BoxTitle = styled.h3`
  margin: 0;
`;

class TitleCard extends React.PureComponent {
  render() {
    const { icon, image, title, url } = this.props;

    return (
      <Box>
        <BoxLink href={url}>
          <BoxArt src={image} alt={title} />
          <BoxBar>
            <BoxTitle>
              {icon && <Icon src={icon} />}
              {title}
            </BoxTitle>
          </BoxBar>
        </BoxLink>
      </Box>
    );
  }
}

TitleCard.propTypes = {
  icon: PropTypes.string,
  image: PropTypes.string,
  title: PropTypes.string,
  url: PropTypes.string,
};

TitleCard.defaultProps = {
  image: '',
};

export default TitleCard;
