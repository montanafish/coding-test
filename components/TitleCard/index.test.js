/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { getShowTitles } from 'api';
import TitleCard from './index';

const showTitle = getShowTitles()[0];

describe('<TitleCard />', () => {
  it('should render a TitleCard__Box', () => {
    const wrapper = shallow(
      <TitleCard
        key={showTitle.id}
        title={showTitle.title}
        image={showTitle.image.medium}
        icon={showTitle.icon}
      />
    );
    expect(wrapper.name()).toBe('TitleCard__Box');
  });

  it('renders children and hide icon', () => {
    const component = renderer.create(
      <TitleCard
        key={showTitle.id}
        title={showTitle.title}
        image={showTitle.image.medium}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
