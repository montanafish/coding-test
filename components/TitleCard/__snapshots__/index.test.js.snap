// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleCard /> renders children and hide icon 1`] = `
.c0 {
  position: relative;
}

.c1:hover {
  opacity: 0.8;
}

.c3 {
  position: absolute;
  bottom: 0;
  background: rgba(0,0,0,0.5);
  color: #fff;
  width: 100%;
  padding: 10px;
  height: 0.33;
}

.c2 {
  width: 100%;
}

.c4 {
  margin: 0;
}

<div
  className="c0"
>
  <a
    className="c1"
  >
    <img
      alt="Home and Away"
      className="c2"
      src="/static/images/tile-medium.jpg"
    />
    <div
      className="c3"
    >
      <h3
        className="c4"
      >
        Home and Away
      </h3>
    </div>
  </a>
</div>
`;
